extensions/binding/org.eclipse.smarthome.binding.iota/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bindingId="iota" xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+   <!-- IOTA Binding -->
+   <thing-type id="topic" extensible="text,number,percentage,onoff">
+      <supported-bridge-type-refs>
+          <bridge-type-ref id="bridge"/>
+      </supported-bridge-type-refs>
+      <label>Topic</label>
+      <description>This thing represents a topic</description>
+      <config-description>
+         <parameter name="root" type="text" required="true">
+            <label>Root Address</label>
+            <description>The root address from which item states should be retrieved from</description>
+         </parameter>
+         <parameter name="refresh" type="integer" min="5">
+            <label>Refresh interval</label>
+            <description>Specifies the refresh interval in seconds.</description>
+         </parameter>
+         <parameter name="mode" type="text" required="true">
+            <options>
+                <option value="public">Public</option>
+                <option value="private">Private</option>
+                <option value="restricted">Restricted</option>
+            </options>
+            <label>MAM Mode</label>
+            <description>The mode of the channel on which to fetch data</description>
+        </parameter>
+        <parameter name="key" type="text" required="false">
+            <label>Private Key</label>
+            <description>Leave blank for non-restricted mode, otherwise enter the private key you want to use</description>
+         </parameter>
+      </config-description>
+   </thing-type>
+   
+   <channel-type id="text">
+        <item-type>String</item-type>
+        <label>Text value</label>
+        <state readOnly="true"> </state>
+        <config-description>
+            <parameter name="stateTopic" type="text" required="true">
+                <label>IOTA state topic</label>
+                <description>An IOTA topic that this thing will subscribe to, to receive the state (e.g. humidity). If set to ANY, any data matcing the item type will be used as state source</description>
```

Comment:
Typo fixed. Regarding stateTopic: it's only something I made up (see my answer to your comment bellow in the `IotaThingHandler.java` class for explanation).

Reply:
In this case I suggest to remove the parameter and rely on the transformation pattern only. As said, it will not be the usual case that the other part communicating over MAM is also an ESH instance. This holds true for all the other data channels, too.
Its also a good test for the transformation pattern, as your live demo (just pretend there is one) must work afterwards ;-)